test/thermo/consistency.cpp

```diff
@@ -0,0 +1,244 @@
+#include "gtest/gtest.h"
+#include "cantera/thermo/ThermoPhase.h"
+#include "cantera/thermo/ThermoFactory.h"
+#include "cantera/base/Solution.h"
+#include "cantera/base/utilities.h"
+#include <regex>
+
+using namespace std;
+
+namespace Cantera
+{
+
+vector<AnyMap> getStates(const string& name) {
+    static AnyMap cases = AnyMap::fromYamlFile("consistency-cases.yaml");
+    return cases[name]["states"].asVector<AnyMap>();
+}
+
+AnyMap getSetup(const string& name) {
+    static AnyMap cases = AnyMap::fromYamlFile("consistency-cases.yaml");
+    return cases[name]["setup"].as<AnyMap>();
+}
+
+// For more informative output about failing test cases
+std::ostream& operator<<(std::ostream& s, const AnyMap& m)
+{
+    if (m.hasKey("file")) {
+        s << fmt::format("file: {}, phase: {}",
+                         m["file"].asString(), m.getString("phase", "<default>"));
+    } else {
+        s << "\n" << m.toYamlString();
+    }
+    return s;
+}
+
+class TestConsistency : public testing::TestWithParam<std::tuple<AnyMap, AnyMap>>
```

Comment:
Some minimal comments would be good to have as a documentation, especially as this test is assuming YAML input that is formatted a very specific way. While it's not extremely difficult to understand, it does take more time to read the code.

Reply:
This isn't marked "ready for review" for a reason. Obviously this test setup warrants some documentation.